nixops/backends/libvirtd.py

```diff
@@ -185,38 +186,19 @@ def _make_os(defn):
                 "    <cmdline>%s</cmdline>"% defn.cmdline if len(defn.kernel) > 0 else "",
                 '</os>']
 
-
-        domain_fmt = "\n".join([
-            '<domain type="{5}">',
-            '  <name>{0}</name>',
-            '  <memory unit="MiB">{1}</memory>',
-            '  <vcpu>{4}</vcpu>',
-            '\n'.join(_make_os(defn)),
-            '  <devices>',
-            '    <emulator>{2}</emulator>',
-            '    <disk type="file" device="disk">',
-            '      <driver name="qemu" type="qcow2"/>',
-            '      <source file="{3}"/>',
-            '      <target dev="hda"/>',
-            '    </disk>',
-            '\n'.join([iface(n) for n in defn.networks]),
-            '    <graphics type="vnc" port="-1" autoport="yes"/>' if not defn.headless else "",
-            '    <input type="keyboard" bus="usb"/>',
-            '    <input type="mouse" bus="usb"/>',
-            defn.extra_devices,
-            '  </devices>',
-            defn.extra_domain,
-            '</domain>',
-        ])
-
-        return domain_fmt.format(
-            self._vm_id(),
-            defn.memory_size,
-            qemu,
-            self._disk_path(defn),
-            defn.vcpu,
-            defn.domain_type
+        result = defn.domain_tpl.format(
+            name=self._vm_id(),
```

Comment:
Here I tried to replace the `memory_size=defn.memory_size,` by `**defn` (more elegant) but couldn't succeed.